### PR TITLE
feat: support disabling CreateVolume in CSI migration

### PR DESCRIPTION
**Pull request by andyzhangx**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
feat: support disabling CreateVolume in CSI migration

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
feat: support disabling CreateVolume in CSI migration
```

### CODE REVIEW

File: pkg/azuredisk/fake_azuredisk_v2.go [hunk 1]
```diff
@@ -20,20 +20,23 @@ package azuredisk
...
 	csicommon "sigs.k8s.io/azuredisk-csi-driver/pkg/csi-common"
 	"sigs.k8s.io/azuredisk-csi-driver/pkg/mounter"
+	"sigs.k8s.io/azuredisk-csi-driver/pkg/optimization"
 	volumehelper "sigs.k8s.io/azuredisk-csi-driver/pkg/util"
+	azcache "sigs.k8s.io/cloud-provider-azure/pkg/cache"
 	azure "sigs.k8s.io/cloud-provider-azure/pkg/provider"
 )
 
 // NewFakeDriver returns a driver implementation suitable for use in unit tests.
-func NewFakeDriver(t *testing.T) (FakeDriver, error) {
```
Comment (edreed) [inline pkg/azuredisk/fake_azuredisk_v2.go:35]: These changes are going to make merges to `main_v2` more difficult because the `Driver` and `DriverV2` structs are very different there. Is this change really necessary?

**Review by edreed — CHANGES_REQUESTED** [review]

File: pkg/azuredisk/azuredisk.go [hunk 2]
```diff
@@ -98,6 +98,7 @@ const (
 	networkAccessPolicyField = "networkaccesspolicy"
 	diskAccessIDField        = "diskaccessid"
 	enableBurstingField      = "enablebursting"
+	csiMigrationField        = "csi-migration"
```
Comment (edreed) [inline pkg/azuredisk/azuredisk.go:101]: Is this field added to the parameters passed to `CreateVolume` by the CSI Migration code? I couldn't find documentation on this.

**Comment by andyzhangx** [comment]
This PR is not necessary since `InTreePluginAzureDiskUnregister` feature gate should work
